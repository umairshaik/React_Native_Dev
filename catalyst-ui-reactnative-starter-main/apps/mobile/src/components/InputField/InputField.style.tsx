import {TextInput, View} from 'react-native';
import styled from 'styled-components/native';

export const StyledText = styled(TextInput).attrs(props => ({
  placeholderTextColor: props.theme.colors.outline,
  color: props.theme.colors.tertiary,
}))`
  height: 43px;
  font-size: 14px;
  background-color: #fff;
  margin: 5px;
`;
export const Container = styled(View)`
  border-color: #aaa;
  border-radius: 5px;
  border-width: 1px;
  flex-direction: row;
  background-color: #fff;
  align-items: center;
  height: 50px;
`;
export const Contain = styled(View)`
  align-self: center;
  padding-left: 10px;
`;
