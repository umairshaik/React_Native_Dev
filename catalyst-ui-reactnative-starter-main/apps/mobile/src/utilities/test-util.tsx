import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
