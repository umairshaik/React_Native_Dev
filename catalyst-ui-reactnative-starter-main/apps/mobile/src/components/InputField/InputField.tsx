import {TextInputProps} from 'react-native';
import {StyledText, Container, Contain} from './InputField.style';

type AdditionalProps = {
  logo?: React.ReactNode;
  accessiblilityProps: string;
};
type Props = AdditionalProps & TextInputProps;

const InputField = ({
  logo,
  style,
  placeholder,
  value,
  accessiblilityProps,
  onChangeText,
  secureTextEntry = false,
}: Props) => (
  <Container style={style}>
    <Contain>{logo}</Contain>
    <StyledText
      testID={accessiblilityProps}
      accessibilityLabel={accessiblilityProps}
      accessibilityHint={`enter your ${accessiblilityProps}`}
      placeholder={placeholder}
      style={style}
      value={value}
      secureTextEntry={secureTextEntry}
      autoCapitalize="none"
      onChangeText={onChangeText}
      editable
    />
  </Container>
);

export default InputField;
