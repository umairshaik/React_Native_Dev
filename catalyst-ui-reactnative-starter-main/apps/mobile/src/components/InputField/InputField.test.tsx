import {Text} from 'react-native';
import InputField from './InputField';
import {render} from '../../utilities/test-util';

describe('[Component] - [InputField] ', () => {
  const mockLogo = () => <Text> hello</Text>;
  test('should render text input correctly', () => {
    const input = render(
      <InputField
        placeholder="Username"
        logo={mockLogo()}
        accessiblilityProps="dummy-button"
        value="dummy"
        onChangeText={() => jest.fn()}
      />,
    );
    expect(input).toMatchSnapshot();
  });
});
