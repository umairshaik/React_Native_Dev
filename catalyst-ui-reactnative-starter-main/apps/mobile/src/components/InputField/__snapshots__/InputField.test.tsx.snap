// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - [InputField]  should render text input correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#aaa",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignSelf": "center",
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text>
       hello
    </Text>
  </View>
  <TextInput
    accessibilityHint="enter your dummy-button"
    accessibilityLabel="dummy-button"
    autoCapitalize="none"
    color="#1C1B1F"
    editable={true}
    onChangeText={[Function]}
    placeholder="Username"
    placeholderTextColor="#79747E"
    secureTextEntry={false}
    style={
      [
        {
          "backgroundColor": "#fff",
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
        },
      ]
    }
    testID="dummy-button"
    value="dummy"
  />
</View>
`;
